sideboard/lib/_services.py

```diff
@@ -11,8 +12,10 @@ def __init__(self, services, name):
         self.services, self.name = services, name
 
     def __getattr__(self, method):
+        from sideboard.lib import is_listy
         assert self.name in self.services, '{} is not registered as a service'.format(self.name)
         service = self.services[self.name]
+        assert not is_listy(getattr(service, '__all__', None)) or method in service.__all__, 'unable to call non-whitelisted method {}.{}'.format(self.name, method)
```

Comment:
This is a security thing.  When exposing a service, you can expose any kind of object.  The two most common ways to do this are to either expose a class instance or just expose a module.

The problem with exposing a whole module is you might end up accidentally including a method you DON'T want clients to call.  To solve this, we check to see if there's a ``__all__`` defined for that module.  If there is, we don't allow methods to be called unless the method is in that list.